feat(tracemetrics): Update table implementations and fix metric name

k-fish opened the pull request:
### Summary
Splits out tables and cleans up some code before infinite table + trace view changes.

k-fish committed:
feat(tracemetrics): Update table implementations and fix metric name

Splits out tables and cleans up some code before infinite table + trace
view changes.

Zylphrex commented on static/app/views/explore/hooks/useMetricOptions.tsx:
```diff
@@ -4,51 +4,59 @@ import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilte
       }
     });
+  } else {
+    query.statsPeriod = '1h'; // Default to a much smaller time window if not searching.
```
I'm inclined to say we should raise this to say 24 hours to capture metrics that are emitted hourly

k-fish committed:
Fix tests

k-fish committed:
fix(tests): Update metric tests to match new API behavior

- Update useMetricOptions tests to match new API (no orderby, includes metric.unit field, client-side sorting)
- Fix useMetricSamplesTable hook to properly pass enabled parameter to useProgressiveQuery
- Remove disableExtrapolation option that was causing double API calls

k-fish committed:
Fix sort

k-fish committed:
knip

Zylphrex commented on static/app/views/explore/metrics/metricsQueryParamsProvider.tsx:
```diff
@@ -0,0 +1,36 @@
+    <MetricsAnalyticsPageSourceContext value="metrics">
+      <MetricsFrozenContextProvider {...freeze}>
+        <MultiMetricsQueryParamsProvider>{children}</MultiMetricsQueryParamsProvider>
+      </MetricsFrozenContextProvider>
```
This doesn't seem quite right. The frozen contexts are per metric right? the `MultiMetricsQueryParamsProvider` contains the data for all the metric on the page.

I think you want to inject it somewhere in [here](https://github.com/getsentry/sentry/blob/master/static/app/views/explore/metrics/metricsQueryParams.tsx#L73-L85) so it's per metric and can contains all the matching trace ids for that metric.

k-fish committed:
24h limit

k-fish committed:
Fixing row styling

k-fish committed:
Fix

k-fish committed:
Fix metricsUrl

k-fish committed:
Fix selector

k-fish committed:
Fix padding and border

k-fish committed:
Bunch of subtle css changes
